lib/collection/proxy-config.js

```diff
@@ -107,6 +107,31 @@ _.assign(ProxyConfig.prototype, /** @lends ProxyConfig.prototype */ {
     },
 
     /**
+     * Used to generate the match regex object from the match string we have.
+     * @returns Match regex object
+     */
+    validateMatchPattern: function(matchPattern) {
+        // No need to match if it is not a valid string.
+        if (!_.isString(matchPattern) || _.isEmpty(matchPattern)) {
+            return ({});
```

Comment:
@czardoz 
For _non string and empty string matchPattern_, i don't want to go into the **.match** stage which is costlier and anyway we knows that it will fails.
Also, IMO i preferred to have the same return type and can be checked _isEmpty_ for **falseness**.